ui/src/reducers/serviceBindings.js

```diff
@@ -9,18 +9,91 @@ import {
   SERVICE_BINDING_DELETE_SUCCESS,
   SERVICE_BINDING_DELETE_FAILURE
 } from '../actions/serviceBinding';
-import resourceReducer from './resource';
 
-const serviceBindings = resourceReducer({
-  readRequest: SERVICE_BINDINGS_REQUEST,
-  readSuccess: SERVICE_BINDINGS_SUCCESS,
-  readFailure: SERVICE_BINDINGS_FAILURE,
-  createRequest: SERVICE_BINDING_CREATE_REQUEST,
-  createSuccess: SERVICE_BINDING_CREATE_SUCCESS,
-  createFailure: SERVICE_BINDING_CREATE_FAILURE,
-  deleteRequest: SERVICE_BINDING_DELETE_REQUEST,
-  deleteSuccess: SERVICE_BINDING_DELETE_SUCCESS,
-  deleteFailure: SERVICE_BINDING_DELETE_FAILURE
-});
+const defaultState = {
+  isFetching: false,
+  boundServices: [],
+  unboundServices: [],
+  errors: [],
+  isCreating: false,
+  isDeleting: false,
+  isActioning: false,
+  isReading: false
+};
 
-export default serviceBindings;
+const getErrors = (error, type, errors) => {
+  const index = errors.findIndex(e => e.type === type);
+  if (!error) {
+    if (index >= 0) {
+      return [...errors.slice(0, index), ...errors.slice(index + 1)];
+    }
+    return errors;
+  }
+  if (index >= 0) {
+    return [...errors.slice(0, index), { error, type }, ...errors.slice(index + 1)];
+  }
+  return [{ error, type }, ...errors];
+};
+
+const serviceBindingsReducer = (state = defaultState, action) => {
+  switch (action.type) {
+    case SERVICE_BINDINGS_REQUEST:
+      return {
+        ...state,
+        isReading: true
+      };
+    case SERVICE_BINDINGS_SUCCESS:
+      return {
+        ...state,
+        isReading: false,
+        errors: getErrors(null, 'read', state.errors),
+        boundServices: action.result.boundServices,
+        unboundServices: action.result.unboundServices
+      };
+    case SERVICE_BINDINGS_FAILURE:
+      return {
+        ...state,
+        isReading: false,
+        errors: getErrors(action.error, 'read', state.errors)
+      };
+    case SERVICE_BINDING_CREATE_REQUEST:
+      return {
+        ...state,
+        isCreating: true
+      };
+    case SERVICE_BINDING_CREATE_SUCCESS:
+      return {
+        ...state,
+        isCreating: false,
+        errors: getErrors(null, 'create', state.errors)
+      };
```

Comment:
when a bind is created, even though it is still being progress on the backend, the UI state should be updated: there is now a new binding, and its status is ""inprogress", the UI state should reflect this change.